### PR TITLE
tests: Add note about implicit validation layers

### PR DESCRIPTION
**Pull request by artem-lunarg**
Add section to tests README.

~~Update VS debugging environment to exclude known implicit layers that affect validation output.~~

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -104,7 +104,12 @@ if (MSVC AND CMAKE_SIZEOF_VOID_P EQUAL 4)
 endif()
 
 if(MSVC_IDE)
-    set_target_properties(vk_layer_validation_tests PROPERTIES VS_DEBUGGER_ENVIRONMENT "VK_LAYER_PATH=$<TARGET_FILE_DIR:VkLayer_khronos_validation>")
+    string(JOIN "\n" CUSTOM_VS_ENVIRONMENT
+        "VK_LAYER_PATH=$<TARGET_FILE_DIR:VkLayer_khronos_validation>"
+        # Disable implicit layers that affect the validation message output
+        "VK_LOADER_LAYERS_DISABLE=VK_LAYER_OBS_HOOK"
```
Comment (juan-lunarg) [inline tests/CMakeLists.txt:110]: I'm not a fan of this approach. Modifying the CMake to be aware of a specific application is strange to read and ages poorly.

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
We have had implicit layers cause problems in the past but AFAIK never came up with a one size fits all approach.

FWIW:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/4b5a19722073afa2394e51092d82a4a655912279

The added documentation looks fine though.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (ncesario-lunarg) [reply tests/CMakeLists.txt:110]: I personally agree with Juan here. I would prefer leaving the cmake as is. Yes folks will have to update their VS projects every time they re-generate cmake (although maybe VS has something similar to VSCode's "launch.json" that can be used to persist these settings?), but I think that is preferable to adding workarounds in cmake.

**Review by ncesario-lunarg — COMMENTED** [review]
The documentation LGTM, but I would also prefer leaving the workaround out of cmake.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (artem-lunarg) [reply tests/CMakeLists.txt:110]: I agree, documentation should give enough information how handle this issue.

**Review by ncesario-lunarg — APPROVED** [review]

File: tests/CMakeLists.txt (hunk 1 above)
Comment (artem-lunarg) [reply tests/CMakeLists.txt]: > Yes folks will have to update their VS projects every time they re-generate cmake.

It should be fine; manually entered debug environment settings are stored in the `*.user` file and preserved when build system is re-generated.

**Pull request merged**